### PR TITLE
Handle TTL for blocked queries seperately 

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

So far, we used `local-ttl=2` hard-coded in dnsmasq's config because this ensured clients won't cache blocked queries for longer times, possibly rendering whitelisting ineffective.

This PR implements a **new TTL for blocked queries only**. This allows users to set `local-ttl` to any value they prefer without this being overwritten in `01-pihole.conf` after each update. This seems useful in context of locally used hostnames that are known to stay constant over long times (e.g., printers, etc.).

The new value defaults to `2` seconds (no change in behavior) and can be controlled using the new FTL config option `BLOCK_TTL=2`

### CODE REVIEW

File: src/dnsmasq_interface.c [hunk 1]
```diff
@@ -77,6 +77,7 @@ static union all_addr null_addrp = {{ 0 }};
 static enum reply_type force_next_DNS_reply = REPLY_UNKNOWN;
 static int last_regex_idx = -1;
 static struct ptr_record *pihole_ptr = NULL;
+#define HOSTNAME "Pi-hole hostname"
```
Comment (dschaper) [inline src/dnsmasq_interface.c:80]: Is this from the hostname PR or is this related to this PR?

Comment (DL6ER) [reply src/dnsmasq_interface.c:80]: It is intended here, now that we need the string in two places, I made it a preprocessor define.

Comment (dschaper) [reply src/dnsmasq_interface.c:80]: Great.

**Review by dschaper — APPROVED** [review]

**Comment by yubiuser** [comment]
Documentation needed.

**Comment by DL6ER** [comment]
See https://github.com/pi-hole/docs/pull/573/commits/60195eadd304ccb0da05d50c3e18cfca45bc20cb in https://github.com/pi-hole/docs/pull/573
